eng/pipelines/templates/steps/build-and-test.yml

```diff
@@ -252,13 +252,31 @@ steps:
     - template: /eng/pipelines/templates/steps/retain-test-logs.yml
     - template: /eng/pipelines/templates/steps/retain-heap-dump-hprofs.yml
 
+  # The name of a test case in the results file isn't helpful in the web view.
+  # Update it to include the breadcrumb namespace and the type.
+  - pwsh: |
+      echo "Current directory: $(get-location)"
+      Get-ChildItem "TEST-*.xml" | ForEach-Object {
+      echo "Updating $($_.Name)"
+      [xml]$xml = Get-Content $_.FullName
+      $xml.SelectNodes("//testcase") | ForEach-Object {
+          $parts = $_.classname -split '\.'
+          $shortClassname = (($parts[0..($parts.Length -2)] | ForEach-Object { $_[0] }) -join ".") + ".$($parts[-1])"
+          $_.name = "$shortClassname.$($_.name)"
+          }
+          $xml.Save($_.FullName)
+      }
+    displayName: 'Update test case names for reporting'
+    workingDirectory: $(System.DefaultWorkingDirectory)/sdk
```

Comment:
Looking at the DevOps Test tab I don't think this did what we wanted. I'm pretty sure this needs to have the `-Recurse` added.

Also, maybe this is something we can do externally to the pipeline so if tests are ever reported to other systems they show differently there too? Maybe have this be a script that gets called during Maven build? This could be even better as it'll collocate the test run and search for reports.

Reply:
I'm gonna leave it here - I worry there maybe other tools folks use that might read this and I don't want to impact them. Good catch on `-Recurse`.